public/app/features/alerting/unified/components/rules/Filter/useRuleFilterAutocomplete.ts

```diff
@@ -7,181 +7,191 @@ import { ComboboxOption } from '@grafana/ui';
-          const truncatedName =
-            namespaceName.length > maxLength ? `${namespaceName.substring(0, maxLength)}...` : namespaceName;
-          const truncatedDescription =
-            namespaceName.length > maxDescriptionLength
-              ? `${namespaceName.substring(0, maxDescriptionLength)}...`
-              : namespaceName;
-          externalNamespaces.push({ label: truncatedName, value: namespaceName, description: truncatedDescription });
-        }
-      });
-    });
-
-    const collator = new Intl.Collator();
-    grafanaFolders.sort((a, b) => collator.compare(a.label ?? '', b.label ?? ''));
-    externalNamespaces.sort((a, b) => collator.compare(a.label ?? '', b.label ?? ''));
-
-    return [...grafanaFolders, ...externalNamespaces];
-  }, [grafanaPromRules, externalPromRulesQueries]);
-
-  const allGroupNames = useMemo(() => {
-    const groupSet = new Set<string>();
-    grafanaPromRules.forEach((namespace) => {
-      namespace.groups.forEach((group) => groupSet.add(group.name));
-    });
-    externalPromRulesQueries.forEach((query) => {
-      query.currentData?.data?.groups?.forEach((group) => {
-        groupSet.add(group.name);
-      });
-    });
-    return Array.from(groupSet).sort();
-  }, [grafanaPromRules, externalPromRulesQueries]);
-
-  const namespacePlaceholder = useMemo(() => {
-    if (isLoadingNamespaces) {
-      return t('common.loading', 'Loading...');
-    }
-    if (namespaceOptions.length === 0) {
-      return t('alerting.rules-filter.no-namespaces', 'No folders available');
-    }
-    return t('alerting.rules-filter.filter-options.placeholder-namespace', 'Select namespace');
-  }, [isLoadingNamespaces, namespaceOptions.length]);
-
-  const groupPlaceholder = useMemo(() => {
-    if (isLoadingNamespaces) {
-      return t('common.loading', 'Loading...');
-    }
-    if (allGroupNames.length === 0) {
-      return t('alerting.rules-filter.no-groups', 'No groups available');
-    }
-    return t('grafana.select-group', 'Select group');
-  }, [isLoadingNamespaces, allGroupNames.length]);
+  const allGroupNames: string[] = [];
+  const isLoadingNamespaces = false;
+  const namespacePlaceholder = t('alerting.rules-filter.filter-options.placeholder-namespace', 'Select namespace');
+  const groupPlaceholder = t('grafana.select-group', 'Select group');
 
   return { namespaceOptions, allGroupNames, isLoadingNamespaces, namespacePlaceholder, groupPlaceholder };
 }
 
 export function useLabelOptions(): {
-  labelOptions: Array<ComboboxOption<string>>;
-  isLoadingGrafanaLabels: boolean;
+  labelOptions: (inputValue: string) => Promise<Array<ComboboxOption<string>>>;
 } {
-  const { labels: grafanaLabels, isLoading: isLoadingGrafanaLabels } =
-    useGetLabelsFromDataSourceName(GRAFANA_RULES_SOURCE_NAME);
+  // Use lazy queries so we only fetch when the dropdown is opened or the user types
+  const [fetchPromNamespaces] = alertRuleApi.useLazyPrometheusRuleNamespacesQuery();
```

Comment:
I think we should use the same api as for 
```
const grafanaResponse = await fetchGrafanaGroups({ limitAlerts: 0, groupLimit: 1000 }).unwrap();
```

If we provide the same arguments, RTKQ could cache requests between the two (it needs an additional, `true` option passed to the fetch trigger)